File: merge-sorted-array/simple.c
// Simple solution: Combine two arrays into the one and sort the elements in increasing order
#include <stdio.h>

void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n){
    int i = m;
    for (;i < nums1Size; i++){
        nums1[i] = nums2[nums1Size - i - 1];
    }
}

// The complexity is based on the sort algorithm, I will skip the optimization for this ques
void sort(int* nums, int numSize){
    int i, j = 0;
    for (i = 0; i < numSize; i++) {
        for (j = i+1; j < numSize; j++) {
            if (nums[i] > nums[j]) {
                int tmp = nums[i];
                nums[i] = nums[j];
                nums[j] = tmp;
            }
        }
    }
}

int main(int argc, char const *argv[])
{
    int m = 3, n = 3;
    int nums1[] = {1,2,3,0,0,0},
        nums2[] = {2,5,6};
    merge(nums1, m+n, m, nums2, n, n);
    sort(nums1, n+m);
    int i;
    for (i = 0; i < m+n; i++) {
        printf("%d\n", nums1[i]);
    }
    return 0;
}

File:  remove-duplicates-from-sorted-array/simple.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	k := 1
	i := 0
	numSize := len(nums)
	prev := nums[0]
	for i < numSize {
		if i > 0 {
			prev = nums[i-1]
		}
		if prev != nums[i] {
			nums[k] = nums[i]
			k++
		}
		i++
	}
	return k
}

func main() {
	nums := []int{-1, 0, 0, 0, 0, 3, 3}
	k := removeDuplicates(nums)
	fmt.Println(k)
	fmt.Println(nums)
}
